src/documentProviders/schema.ts

```diff
@@ -4,36 +4,51 @@ import { SchemaString, SchemasV1Api } from "../clients/schemaRegistryRest";
 import { Schema } from "../models/schema";
 import { getSidecar } from "../sidecar";
 
+/** Makes a read-only editor buffer holding a schema */
 export class SchemaDocumentProvider extends ResourceDocumentProvider {
+  // non-file, non-untitled URIs cause the resulting buffer to be read-only
   scheme = "confluent.schema";
 
   public async provideTextDocumentContent(uri: vscode.Uri): Promise<string> {
+    // parse the query string into the JSON respresentation of the original schema object
     const schemaObj: Schema = this.parseUriQueryBody(uri.query) as Schema;
-    // recreate the Schema instance so we can access the .connectionId getter
+
+    // Promote from as-from-json to a full Schema object
     const schema = Schema.create(schemaObj);
-    // fetch the schema definition from the sidecar and attempt to prettify it before displaying
-    const client: SchemasV1Api = (await getSidecar()).getSchemasV1Api(
-      schema.schemaRegistryId,
-      schema.connectionId,
-    );
-    const schemaResp: SchemaString = await client.getSchema({ id: Number(schema.id) });
-    const schemaDefinition: string | null = prettifySchemaDefinition(schemaResp);
-    if (!schemaDefinition) {
-      throw new Error("Failed to load schema definition; it may be empty or invalid.");
-    }
-    return schemaDefinition;
+
+    // Get the prettified schema definition
+    return fetchSchemaBody(schema);
   }
 }
 
+/**
```

Comment:
Factor out from` SchemaDocumentProvider` so that evolve schema action can also use.